UnitTests/TestVTableSwapHook2.cpp

```diff
@@ -0,0 +1,66 @@
+#include <memory>
+#include <stdexcept>
+
+#include <Catch.hpp>
+
+#include "polyhook2/Virtuals/VTableSwapHook.hpp"
+#include "polyhook2/Tests/StackCanary.hpp"
+
+// original class
+
+class MyClass {
+public:
+	virtual ~MyClass() {}
+
+	virtual int method1(int x) {
+		return 2 * x;
+	}
+
+	virtual int method2(int x, int y) {
+		return x + y;
+	}
+};
+
+// virtual function hooks
+
+int myclass_method1(MyClass* pThis, int x);
+int myclass_method2(MyClass* pThis, int x, int y);
+
+// helper typedefs, and unique_ptr for storing the hook
+
+typedef PLH::VFunc<1, decltype(&myclass_method1)> VMethod1;
+typedef PLH::VFunc<2, decltype(&myclass_method2)> VMethod2;
+typedef PLH::VTableSwapHook2<MyClass, VMethod1, VMethod2> VTableMyClass;
+std::unique_ptr<VTableMyClass> hook = nullptr;
+
+// hook implementations
+
+NOINLINE int myclass_method1(MyClass* pThis, int x) {
+	if (!hook)
+		throw std::runtime_error("original function not hooked");
```

Comment:
Please remove all exceptions and use the TestEffectTracker.trigger() like other tests do.

Reply:
Yes, done in the latest patch.